spec/stack.js

```diff
@@ -13,10 +13,52 @@ describe('Stack', () => {
 
   context('push()', () => {
     it('pushes an element to the top of the stack.', () => {
-      const myStack = new Stack()
+      const stack = new Stack()
 
-      expect(() => myStack.push('foo'))
-        .to.alter(() => myStack.length(), { from: 0, to: 1 })
+      expect(() => stack.push('foo'))
+        .to.alter(() => stack.length(), { from: 0, to: 1 })
+    })
+
+    it('pushes an element to the top of the stack when there is preexisting data', () => {
+      const stack = new Stack()
+      stack.push('alice')
+      stack.push('bob')
+
+      expect(() => stack.push('foo'))
+        .to.alter(() => stack.peek(), {from: 'bob', to: 'foo'})
+    })
+
+  })
+  context('pop()', () => {
```

Comment:
I would use `describe` instead of `context` they're litterally the same function but `describe` is for a thing and `context` is for a specific `state of the world` where things should be different. 

```js
describe('Stack', function(){
  describe('#push', function(){
    
    let stack;

    context('when we have 0 items in our stack', function(){
      beforeEach(function(){
        stack = new Stack()
      })
      it('should add the given item to the stack', function(){
        expect(() => stack.push('foo'))
          .to.alter(() => stack.length(), {from: 0, to: 1})
      })
    })

    context('when we have 3 items in our stack', function(){
      beforeEach(function(){
        stack = new Stack()
        stack.push('a')
        stack.push('b')
        stack.push('c')
      })
      it('should add the given item to the stack', function(){
        expect(() => stack.push('foo'))
          .to.alter(() => stack.length(), {from: 3, to: 4})
      })
    })

  })
})

```